test/runtime_errors.js

```diff
@@ -76,4 +82,10 @@ describe("Runtime Errors", function() {
     });
   });
 
+  after('shutdown', function(done) {
+    let provider = web3._provider;
+    web3.setProvider();
+    provider.close(done);
```

Comment:
What does `web3.setProvider()` do when not passing in a provider as a parameter?

Reply:
from memory it sets the provider to `undefined` (it doesn't check the arg). I think I was being overly pedantic here though. My thought process was that web3 has a bunch of stuff it does in the background, and I didn't want it trying to run a request during or after provider close.

Further, I think calling `provider.close` is a good idea in general when using the direct provider (probably need to make sure other tests do this), but unfortunately it's not a standard part of the interface, so I need to resort to something like `provider.connection.close` for the WebSocket provider. Maybe I'll raise an issue for that.